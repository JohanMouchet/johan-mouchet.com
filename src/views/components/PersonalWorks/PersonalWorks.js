// @flow

import * as React from "react";
import cx from "classnames";
import "./___.scss";

/**
 * Personal Works
 *
 * @param {array}   works                  - List of works
 * @param {boolean} works[0].video         - True if work type is video
 * @param {array}   works[0].thumbnailPath - Work thumbnail image path
 * @param {array}   works[0].expandedPath  - Work expanded image path
 */

const ___ = (props: Props) => {
  const {
    prop1,
    prop2
  } = props;

  const ___Class = cx(
    "___",
    {
      prop1: "___--prop1"
    }
  );

  return (
	<div className="c-personal-works">
		<div className="grid">
			{% for work in options.works %}
				<div className="cell cell--6-@xs cell--4-@sm cell--2-@md">
					<section className="c-personal-work">
						<a className="c-personal-work__overlay" href="{ work.URL }" target="_blank">
							<img className="c-personal-work__thumbnail" src="{ work.thumbnailPath }" alt="Personal work"/>
							{% if work.sourceIconPath %}
								<span className="c-personal-work__source">
									{ file_get_contents(work.sourceIconPath)|raw }
								</span>
							{% endif %}
						</a>
					</section>
				</div>
			{% endfor %}
		</div>
	</div>
  );
};

___.defaultProps = {
  prop1: 2
};

export default ___;
