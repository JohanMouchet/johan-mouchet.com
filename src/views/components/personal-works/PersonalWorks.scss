@import "assets/styles/settings/_settings";
@import "assets/styles/tools/_tools";

.c-personal-work {
  $root: &;

  &__overlay {
    display: block;
    position: relative;
    max-width: 200px;

    &::after {
      content: "";
      position: absolute;
      top: 0;
      left: 0;
      height: 100%;
      width: 100%;
      background-image: map-get($s-backgrounds, dots),
        linear-gradient(
          rgba($s-color-primary-lighter, 0.7),
          rgba($s-color-primary-darker, 0.9)
        );
      transform-origin: 100% 50%;
      transition: transform 0.2s ease-out;
    }

    &:hover,
    &:focus {
      &::after {
        transform: scaleX(0);
      }

      #{$root}__thumbnail {
        filter: grayscale(0);
      }

      #{$root}__source {
        opacity: 0;
      }
    }
  }

  &__thumbnail {
    display: block;
    border: 1px solid $s-color-gray-lighter;
    filter: grayscale(1);
    transition: filter 0.2s ease-out;
  }

  &__source {
    line-height: 0;
    position: absolute;
    bottom: 1em;
    right: 1em;
    z-index: map-get($s-layers, personal-work);
    transition: opacity 0.2s ease-out;

    svg {
      height: t-modular-scale(1);
      width: t-modular-scale(1);
      fill: $s-color-white;
    }
  }
}
