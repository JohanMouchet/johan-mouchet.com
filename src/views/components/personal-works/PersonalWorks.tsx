import { IconUnsplash, IconYoutube } from "@/views/objects/icons";
import clsx, { ClassValue } from "clsx";
import styles from "./PersonalWorks.module.scss";

export const PersonalWorks = ({
  works,
  className,
  ...props
}: {
  works: Array<{
    link: {
      url: string;
    };
    thumbnailSrc: {
      url: string;
    };
    thumbnailExternalSrc: {
      url: string;
    };
    icon: "unsplash" | "youtube";
  }>;
  className?: ClassValue;
} & React.HTMLProps<HTMLDivElement>) => {
  const getIcon = (icon: string) => {
    if (icon === "unsplash") {
      return <IconUnsplash />;
    } else if (icon === "youtube") {
      return <IconYoutube />;
    }
  };

  return !works?.length ? null : (
    <div className={clsx(styles["c-personal-works"], className)} {...props}>
      <div className="grid">
        {works.map((work) => (
          <div
            className="cell cell-6 sm:cell-4 md:cell-2 u-vr--bottom-1-@xs u-vr--bottom-0-@md"
            key={work.link.url}
          >
            <section className={styles["c-personal-work"]}>
              <a
                className={styles["c-personal-work__overlay"]}
                href={work.link.url}
              >
                {/* eslint-disable-next-line @next/next/no-img-element */}
                <img
                  className={styles["c-personal-work__thumbnail"]}
                  src={work.thumbnailSrc?.url || work.thumbnailExternalSrc?.url}
                  alt={`Personal work on ${work.icon}`}
                  loading="lazy"
                />
                {work.icon && (
                  <span
                    className={styles["c-personal-work__source"]}
                    aria-hidden
                  >
                    {getIcon(work.icon)}
                  </span>
                )}
              </a>
            </section>
          </div>
        ))}
      </div>
    </div>
  );
};
