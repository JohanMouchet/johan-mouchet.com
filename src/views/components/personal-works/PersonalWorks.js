// @flow

import * as React from "react";
import "./PersonalWorks.scss";

type Props = {
  works: [
    {
      video?: boolean,
      url: string,
      thumbnailPath: string,
      sourceIcon: Array<mixed>
    }
  ]
};

const PersonalWorks = (props: Props) => {
  const { works } = props;

  return !works ? null : (
    <div className="c-personal-works">
      <div className="grid">
        {works.map(work => (
          <div
            className="cell cell--6-@xs cell--4-@sm cell--2-@md"
            key={work.thumbnailPath}
          >
            <section className="c-personal-work">
              <a
                className="c-personal-work__overlay"
                href={work.url}
                target="_blank"
                rel="noopener noreferrer"
              >
                <img
                  className="c-personal-work__thumbnail"
                  src={work.thumbnailPath}
                  alt="Personal work"
                />
                {work.sourceIcon && (
                  <span className="c-personal-work__source">
                    {work.sourceIcon}
                  </span>
                )}
              </a>
            </section>
          </div>
        ))}
      </div>
    </div>
  );
};

export default PersonalWorks;
