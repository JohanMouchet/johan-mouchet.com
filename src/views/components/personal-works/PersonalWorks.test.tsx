import React from "react";
import { shallow } from "enzyme";
import PersonalWorks from ".";

it("Renders <PersonalWorks /> without crashing", () => {
  expect(shallow(<PersonalWorks />)).toMatchSnapshot();
});
