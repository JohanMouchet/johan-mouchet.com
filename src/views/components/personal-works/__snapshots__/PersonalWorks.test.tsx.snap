// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <PersonalWorks /> without crashing 1`] = `
<div
  class="c-personal-works"
>
  <div
    class="grid"
  >
    <div
      class="cell cell-6 sm:cell-4 md:cell-2 u-vr--bottom-1-@xs u-vr--bottom-0-@md"
    >
      <section
        class="c-personal-work"
      >
        <a
          class="c-personal-work__overlay"
          href="#"
        >
          <img
            alt="Personal work on unsplash"
            class="c-personal-work__thumbnail"
            loading="lazy"
            src="https://via.placeholder.com/200x200"
          />
          <span
            aria-hidden="true"
            class="c-personal-work__source"
          >
            <svg />
          </span>
        </a>
      </section>
    </div>
  </div>
</div>
`;
